### ISSUE TITLE
how to filter on phone number field?

### ISSUE THREAD

**Issue by littlehome-eugene** [stefanfoulis/django-phonenumber-field#153]
When a user inputs 010-xxxx-xxxx in our country (with country code 82)

we'd like to perform a query with '010xxx' but phonenumberfiled seems to ignore the first leading zero.

```
In [25]: Address.objects.filter(phone__contains='10')[0].phone.as_national
Out[25]: '010-9999-9999'

In [26]: Address.objects.filter(phone__contains='10')[0].id
Out[26]: 2999

In [27]: Address.objects.filter(id=2999).filter(phone__contains='010')
Out[27]: []

```

**Comment by amateja** [comment]
If I recollect right it depends on how you store values in database... which is bad. By default it is stored in international format. Please try `Address.objects.filter(id=2999, phone__contains='+10')` or even `Address.objects.filter(id=2999, phone__startswith='+10')`

**Comment by littlehome-eugene** [comment]
hmm it's ambiguous, when user searches for '010', it could mean +10 or it could literally mean +10-xxxx-010 or somthing similar.

**Comment by amateja** [comment]
That's true. But please remember that under the hood PhonenumberField is actually CharField child class and value is stored in database according to `PHONENUMBER_DB_FORMAT` setting. By default it is E164 a.k.a. international format. The `contains` and `startswith` lookups are default CharField lookups so they look according to theirs default behaviour and looks through data in format defined by your settings.

A side note here: I really thing that it is a bad idea to change it to any other format because you might loose some part of your data. Assume two countries has the same number formats or same area code. Replacing international format with local and setting `PHONENUMBER_DEFAULT_REGION` may cause that number from e.g. Canada will be presented as US one. Loosing data in IT is really, really bad 😉. I've been thinking lately about changing a lot of PhonenumbersField internals... but till now I couldn't find enough time. Keep fingers crossed!

**Comment by amateja** [comment]
To elaborate that even more lookups construct SQL queries and there is no SQL syntax supporting phone number format.

https://github.com/django/django/blob/944610a86c81f90c3c79e5440a5e2c706aa0ef62/django/db/models/lookups.py#L435
contains gives SQL query as `my_field = "%010%"`
https://github.com/django/django/blob/944610a86c81f90c3c79e5440a5e2c706aa0ef62/django/db/models/lookups.py#L457
startswith gives SQL query as `my_field = "010%"`
Both cannot find anything because your value is stores as `+1099999999`. We might add our PhonenumberField specific lookups but it would be very complicated.
